test/utils/InstanceDefinitionUtils.test.ts

```diff
@@ -0,0 +1,170 @@
+
+  const valueSet = new InstanceDefinition();
+  valueSet.id = 'MyVS';
+  valueSet.resourceType = 'ValueSet';
+  valueSet._instanceMeta.name = 'MyVS';
+  valueSet._instanceMeta.usage = 'Definition';
+
+  const operation = new InstanceDefinition();
+  operation.id = 'MyOperation';
+  operation.resourceType = 'OperationDefinition';
+  operation._instanceMeta.name = 'MyOperation';
+  operation._instanceMeta.usage = 'Definition';
+
+  const logicalModel = new InstanceDefinition();
+  logicalModel.id = 'MyLM';
+  logicalModel.resourceType = 'StructureDefinition';
+  logicalModel._instanceMeta.name = 'MyLM';
+  logicalModel._instanceMeta.usage = 'Definition';
+  logicalModel.kind = 'logical';
+
+  const extension = new InstanceDefinition();
+  extension.id = 'MyExtension';
+  extension.resourceType = 'StructureDefinition';
+  extension._instanceMeta.name = 'MyExtension';
+  extension._instanceMeta.usage = 'Definition';
+  extension.kind = 'resource';
+  extension.type = 'Extension';
+
+  const extensionExampleInferred = new InstanceDefinition();
+  extensionExampleInferred.id = 'MyExtension';
+  extensionExampleInferred.resourceType = 'StructureDefinition';
+  extensionExampleInferred._instanceMeta.name = 'MyExtension';
+  extensionExampleInferred.kind = 'resource';
+  extensionExampleInferred.type = 'Extension';
+
+  const profile = new InstanceDefinition();
+  profile.id = 'MyProfile';
+  profile.resourceType = 'StructureDefinition';
+  profile._instanceMeta.name = 'MyProfile';
+  profile._instanceMeta.usage = 'Definition';
+  profile.kind = 'resource';
+  profile.type = 'something-not-extension';
+
+  const primitiveSD = new InstanceDefinition();
+  primitiveSD.id = 'MyPrimitive';
+  primitiveSD.resourceType = 'StructureDefinition';
+  primitiveSD._instanceMeta.name = 'MyPrimitive';
+  primitiveSD._instanceMeta.usage = 'Definition';
+  primitiveSD.kind = 'something-not-extension';
+  primitiveSD.type = 'primitive-type';
+
+  const observation = new InstanceDefinition();
+  observation.id = 'MyObservation';
+  observation.resourceType = 'Observation';
+  observation._instanceMeta.name = 'MyObservation';
+  observation._instanceMeta.usage = 'Definition';
+
+  const instances: InstanceDefinition[] = [
+    examplePatient,
+    capabilityStatement,
+    capabilityStatementExample,
+    valueSet,
+    operation,
+    logicalModel,
+    extension,
+    extensionExampleInferred,
+    profile,
+    primitiveSD,
+    observation
+  ];
+
+  it('should filter out all example instances regardless of resourceType', () => {
+    const examples = filterExampleInstances(instances);
+    expect(examples).toEqual([
+      examplePatient,
+      capabilityStatementExample,
+      extensionExampleInferred
+    ]);
+    expect(instances).toEqual([
+      capabilityStatement,
+      valueSet,
+      operation,
+      logicalModel,
+      extension,
+      profile,
+      primitiveSD,
+      observation
+    ]);
+  });
+  describe('Non-example instances', () => {
```

Comment:
Yeah I had them set up this way in order to mirror what happens in `app.ts` since that file is dependent on the mutation working correctly. I did think about how there is a weird dependency between these tests now, so I agree with you there. I think maybe the right approach is to reset `instances` before each test. That way, the tests can check that the right elements get filtered out but they will also be independent of each other.